docs/introduction/index.md

```diff
@@ -0,0 +1,109 @@
+
+[Razor](https://docs.microsoft.com/aspnet/core/mvc/views/razor) is a syntax for combining HTML markup with C# code. Razor is designed for developer productivity, allowing the developer to switch between markup and C# in the same file with [IntelliSense](https://docs.microsoft.com/visualstudio/ide/using-intellisense) support. The following markup is an example of a basic custom dialog component in a Razor file (*DialogComponent.cshtml*):
+
+```cshtml
+<div>
+    <h2>@Title</h2>
+    @RenderContent(Body)
+    <button onclick=@OnOK>OK</button>
+</div>
+
+@functions {
+    public string Title { get; set; }
+    public Content Body { get; set; }
+    public Action OnOK { get; set; }
+}
+```
+
+When this component is used elsewhere in the app, IntelliSense speeds development with syntax completion and parameter info.
+
+Components can be:
+
+* Nested.
+* Generated procedurally in code.
+* Shared via class libraries.
+* Unit tested without requiring a browser DOM.
+
+## Infrastructure
+
+Blazor offers the core facilities that most apps require, including:
+
+* Layouts
+* Routing
+* Dependency injection
+* Unit testing
+
+All of these features are optional. When one of these features isn't used in an app, the implementation is stripped out of the app when published by the IL linker.
+
+A few low-level elements are included in the framework. For example, routing and layouts aren't built-in. Routing and layouts are implemented in *user space*, code that an app developer can write without using internal APIs. These features can be replaced with different systems to suit the app's requirements. The current layouts prototype is implemented in about 30 lines of C# code, so a developer can understand and replace it if desired.
+
+## Code sharing and .NET Standard
+
+The [.NET Standard](https://docs.microsoft.com/dotnet/standard/net-standard) is a formal specification of .NET APIs that are intended to be available on all .NET implementations. Blazor supports .NET Standard 2.0 or higher. APIs that can't be supported due to browser limitations (for example, accessing the file system, opening a socket, threading, and others) throw [PlatformNotSupportedException](https://docs.microsoft.com/dotnet/api/system.platformnotsupportedexception). .NET Standard class libraries can be shared across server code and in browser-based apps.
+
+## JavaScript/TypeScript interop
+
+For apps that require third-party JavaScript libraries and browser APIs, WebAssembly is designed to interoperate with JavaScript. Blazor is capable of using any library or API that JavaScript is able to use. JavaScript code can call into C# (for example, to handle an event).
+
+## Optimization
+
+For client-side apps, payload size is critical. Blazor optimizes payload size to reduce download times. For example, unused IL code is removed when a Blazor app is built, and response compression is utilized.
+
+## Deployment
```

Comment:
This "deployment" section is a bit weird. I think the goal is to explain the two deployment models, but information about each of them is mixed together. I'd recommend:

## Deployment

You can use Blazor to build a pure standalone client-side application, or a full-stack .NET application containing both server and client.

 * If you build a **standalone client-side application**, then your Blazor app is compiled to a *dist* directory containing nothing but static files. Those files can be hosted on the Azure CDN, GitHub Pages, Node.js servers, and many other servers and services.
 * If you build a **full-stack .NET application**, then you have the additional advantage of easily sharing code between server and client. The resulting ASP.NET Core server application, which serves the Blazor client-side UI as well as any server-side API endpoints, can be built and deployed to any cloud or on-premise host supported by ASP.NET Core.